### ISSUE TITLE
`Layout/EmptyLineAfterGuardClause` flags heredoc inside multi-line `if`.

### ISSUE THREAD

**Issue by reitermarkus** [rubocop-hq/rubocop#6300]
## Expected behavior

No offense should occur and auto-correct should not add a newline inside the `if` statement.

## Actual behavior

An offense occurs and auto-correct adds two newlines inside the `if` statement.

## Steps to reproduce the problem

Running `rubocop` on

```ruby
def method
  if truthy
    raise <<-MSG
      This is an error.
    MSG
  end

  value
end
```

shows 

```
Inspecting 1 file
C

Offenses:

file.rb:5:1: C: Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.
    MSG
^^^^^^^

1 file inspected, 1 offense detected
```

and running `rubocop --auto-correct` produces

```ruby
def method
  if truthy
    raise <<-MSG
      This is an error.
    MSG


  end

  value
end
```

and shows

```
Inspecting 1 file
C

Offenses:

file.rb:5:1: C: [Corrected] Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.
    MSG
^^^^^^^
file.rb:7:1: C: [Corrected] Layout/EmptyLines: Extra blank line detected.

1 file inspected, 2 offenses detected, 2 offenses corrected
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.59.1 (using Parser 2.5.1.2, running on ruby 2.3.7 universal.x86_64-darwin17)
```

**Comment by diachini** [comment]
Thanks @reitermarkus, for making this issue first!  (Was just about to make one myself.)

I am assuming this is a side-effect of PR #6273 (the most recent edge case that was fixed around this) or #6284 (the autocorrect for that edge case).  We were not experiencing this on 0.59.0, but are, now on 0.59.1

**Comment by koic** [comment]
Thanks for the feedback.
Yes. This is a regression at 0.59.1 💦 I opened a PR #6301 that fixed this regression and added some test cases.
